src/plugin/pid/pid_miscwrappers.cpp

```diff
@@ -66,6 +72,82 @@ fork()
   return realPid;
 }
 
+extern VirtualPidTable *virtPidTableInst;
+VirtualPidTable vfork_saved_virtPidTableInst;
+
+LIB_PRIVATE void
+pidVirt_vfork_prepare()
+{
+  vfork_saved_pid = getpid();
+  vfork_saved_ppid = getppid();
+  vfork_saved_tid = dmtcp_gettid();
+  vfork_saved_virtPidTableInst = *virtPidTableInst;
+
+  Util::getVirtualPidFromEnvVar(&childVirtualPid, NULL, NULL);
+}
+
+static pid_t vforkPid = 0;
+static char* stackStart;
+static size_t stackSize;
+static void* newStackAddr;
+
+static void postVforkRestoreStack()
+{
+}
+
+extern "C" pid_t
+vfork()
+{
+  char dummy = 0;
+  //
+  // TODO: Add more comments before committing the code.
```

Comment:
@karya0 , could you do what the comment says?  :-)
`// TODO: Add more comments before committing the code.`